File: app/src/main/java/com/aus/ausgegeben/data/PrefsLww.kt
package com.aus.ausgegeben.data

/** Last-write-wins for `users/{uid}/settings/preferences.updatedAt`. Equal clocks are a no-op. */
enum class PrefsLwwAction {
    APPLY_REMOTE,
    PUSH_LOCAL,
    HOLD,
}

fun prefsLwwAction(remoteUpdatedAt: Long, localUpdatedAt: Long): PrefsLwwAction = when {
    remoteUpdatedAt > localUpdatedAt -> PrefsLwwAction.APPLY_REMOTE
    localUpdatedAt > remoteUpdatedAt -> PrefsLwwAction.PUSH_LOCAL
    else -> PrefsLwwAction.HOLD
}

File: web/src/services/prefsLww.ts
/** Last-write-wins for `users/{uid}/settings/preferences.updatedAt`. Equal clocks are a no-op. */
export type PrefsLwwAction = 'apply_remote' | 'push_local' | 'hold';

export function prefsLwwAction(remoteUpdatedAt: number, localUpdatedAt: number): PrefsLwwAction {
  if (remoteUpdatedAt > localUpdatedAt) return 'apply_remote';
  if (localUpdatedAt > remoteUpdatedAt) return 'push_local';
  return 'hold';
}
